source/features/open-all-notifications.js

```diff
@@ -1,60 +1,106 @@
 import {h} from 'dom-chef';
 import select from 'select-dom';
 import delegate from 'delegate';
-import {groupButtons} from '../libs/utils';
+import * as icons from '../libs/icons';
+import {groupButtons, safeElementReady} from '../libs/utils';
 import {isNotifications} from '../libs/page-detect';
 
-function openNotifications() {
-	const urls = select.all('.unread .js-notification-target').map(el => el.href);
+const unreadNotificationsClass = '.unread .js-notification-target';
+let notificationsToOpen = [];
+
+const faceboxTrigger = (
+	<a href="#open_all_in_tabs" id="facebox-trigger" rel="facebox" style={{display: 'none'}}></a>
+);
+
+function openNotifications(unreadNotifications) {
+	const urls = unreadNotifications.map(el => el.href);
+
 	browser.runtime.sendMessage({
 		urls,
 		action: 'openAllInTabs'
 	});
-	for (const listItem of select.all('.list-group .list-group-item')) {
+
+	for (const notification of unreadNotifications) {
+		const listItem = notification.closest('.list-group-item');
 		listItem.classList.add('read');
+		listItem.classList.remove('unread');
+	}
+}
+
+function tryOpeningNotifications(container = document) {
+	const unreadNotifications = select.all(unreadNotificationsClass, container);
+
+	if (unreadNotifications.length < 5) {
+		openNotifications(unreadNotifications);
+	} else {
+		notificationsToOpen = unreadNotifications;
+		faceboxTrigger.click();
 	}
 }
 
 export default function () {
 	if (!isNotifications() || select.exists('[href="#open_all_in_tabs"]')) {
 		return;
 	}
-	const unreadCount = select.all('.unread .js-notification-target').length;
 
-	if (unreadCount === 0) {
-		return;
-	}
+	faceboxTrigger.addEventListener('click', async () => {
+		await safeElementReady('#facebox-description');
```

Comment:
I don't think this is necessary, the `#facebox-description` element is appended right after this event listener.

Reply:
I've already tested this, `#facebox-description` is not appended into the facebox until it is triggered once.

It is required for the first time, and not needed for repeated calls.